src/main/java/com/google/codeu/servlets/ChartServlet.java

```diff
@@ -48,15 +51,25 @@ public void init() {
 
    @Override
    /**
-    * Converts the returned value to JSON and prints the JSON out as a response.
-    *
-    * @return a list of all of the messages
+    * For page /charts, generates a request with current user and login status,
+    * and forwards that request to chart.jsp
     */
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      response.setContentType("application/json");
-      List<Message> msgList = datastore.getMessages(null);
-      Gson gson = new Gson();
-      String json = gson.toJson(msgList);
-      response.getWriter().println(json);
+    public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+
+      UserService userService = UserServiceFactory.getUserService();
+
+      boolean isUserLoggedIn = userService.isUserLoggedIn();
+      request.setAttribute("isUserLoggedIn", isUserLoggedIn);
+
+      if (isUserLoggedIn) {
+        String user = userService.getCurrentUser().getEmail();
+        request.setAttribute("user", user);
+      }
+
+      List<Message> messages = datastore.getAllMessages();
```

Comment:
I don't think you actually use this for anything?

Reply:
You're right -- this is leftover. Now, the messages are coming in JSON form from /feed-json.